src/Main.elm

```diff
@@ -1,10 +1,362 @@
+                        newModel =
+                            { model | audience = data, errorMessage = Nothing }
+                    in
+                    ( newModel, Cmd.none )
+
+                Err error ->
+                    ( { model | errorMessage = Just (errorMessage error) }, Cmd.none )
+
+        OpenFolder folderId ->
+            ( { model | currentFolderId = Just folderId }, Cmd.none )
+
+        GoBack ->
+            case model.currentFolderId of
+                Just folderId ->
+                    let
+                        newId =
+                            getActualParent folderId model
+                    in
+                    ( { model | currentFolderId = newId }, Cmd.none )
+
+                Nothing ->
+                    ( model, Cmd.none )
+
+        SelectCategory category ->
+            ( { model | selectedCategory = category, currentFolderId = Nothing }, Cmd.none )
+
+
+errorMessage : Http.Error -> String
+errorMessage error =
+    case error of
+        NetworkError ->
+            "Network Error"
+
+        BadUrl _ ->
+            "Bad URL"
+
+        Timeout ->
+            "Timeout"
+
+        BadStatus _ ->
+            "Bad status"
+
+        BadBody reason ->
+            reason
+
+
+
+--Get the parent of the folder, if there is no parent he put the currentFolderId to Nothing so he come back
+--to the inicial screen
+
+
+getActualParent : Int -> Model -> Maybe Int
+getActualParent folderId model =
+    case getFolderById folderId model of
+        Just folder ->
+            case folder.parent of
+                Just parentId ->
+                    Just parentId
+
+                Nothing ->
+                    Nothing
+
+        Nothing ->
+            Nothing
+
+
+getFolderById : Int -> Model -> Maybe AudienceFolder.AudienceFolder
+getFolderById targetId model =
+    List.filter (\folder -> folder.id == targetId) model.audienceFolder |> List.head
+
+
+
+--SUBSCRIPTIONS
+
+
+subscriptions : Model -> Sub Msg
+subscriptions _ =
+    Sub.none
+
+
+
+--INIT
+
+
+init : () -> ( Model, Cmd Msg )
+init _ =
+    ( initModel, Cmd.batch [ cmdSearchFolder initModel, cmdSearchAudience initModel ] )
+
+
+initModel : Model
+initModel =
+    { audienceFolder = []
+    , audience = []
+    , urlAudienceFolder = "http://localhost:8000/server/Json/AudienceFolder.Json"
+    , urlAudience = "http://localhost:8000/server/Json/Audience.Json"
+    , errorMessage = Nothing
+    , currentFolderId = Nothing
+    , selectedCategory = Audience.Authored
+    }
+
+
+
+--CMD
+
+
+cmdSearchFolder : Model -> Cmd Msg
+cmdSearchFolder model =
+    Http.get
+        { url = model.urlAudienceFolder
+        , expect = Http.expectJson MsgGotResultsAudienceFolder Decoder.decodeAudienceFolders
+        }
+
+
+cmdSearchAudience : Model -> Cmd Msg
+cmdSearchAudience model =
+    Http.get
+        { url = model.urlAudience
+        , expect = Http.expectJson MsgGotResultsAudience Decoder.decodeAudiences
+        }
```

Comment:
I'd better rename this to both `cmdFetchFolders` and `cmdFetchAudiences` since that's what your functions seem to do. I don't think _Search_ is a correct term here. But extra points for using HTTP calls in the app 👍🏻